Lib/test/test_re.py

```diff
@@ -2238,6 +2254,34 @@ def test_findall_atomic_grouping(self):
         self.assertEqual(re.findall(r'(?>(?:ab)?)', 'ababc'), ['ab', 'ab', '', ''])
         self.assertEqual(re.findall(r'(?>(?:ab){1,3})', 'ababc'), ['abab'])
 
+    def test_bug_gh91616(self):
```

Comment:
Aren't these test redundant? New assertions in `test_fullmatch_possessive_quantifiers` and `test_fullmatch_atomic_grouping` cover all changes.

Reply:
done